repos/system_upgrade/common/actors/livemode/modify_userspace_for_livemode/files/do-upgrade.sh

```diff
@@ -0,0 +1,364 @@
+        local dirname
+        dirname="$("$NEWROOT/bin/dirname" "$NEWROOT$LEAPP_FAILED_FLAG_FILE")"
+        [ -d "$dirname" ] || mkdir "$dirname"
+        "$NEWROOT/bin/touch" "$NEWROOT$LEAPP_FAILED_FLAG_FILE"
+    fi
+
+    # Dump debug data in case something went wrong
+    ##if want_inband_dump "$rv"; then
+    ##    collect_and_dump_debug_data
+    ##fi
+
+    # NOTE: THIS SHOULD BE AGAIN PART OF LEAPP IDEALLY
+    ## backup old product id certificates
+    #chroot $NEWROOT /bin/sh -c 'mkdir /etc/pki/product_old; mv -f /etc/pki/product/*.pem /etc/pki/product_old/'
+
+    ## install new product id certificates
+    #chroot $NEWROOT /bin/sh -c 'mv -f /system-upgrade/*.pem /etc/pki/product/'
+
+    # restore things twiddled by workarounds above. TODO: remove!
+    if [ -f /sys/fs/selinux/enforce ]; then
+        echo "$enforce" > /sys/fs/selinux/enforce
+    fi
+    return $rv
+}
+
+save_journal() {
+    # Q: would it be possible that journal will not be flushed completely yet?
+    echo "writing logs to disk and rebooting"
+
+    local logfile="/sysroot/tmp-leapp-upgrade.log"
+
+    # Create logfile if it doesn't exist
+    [ -n "$logfile" ] && true > $logfile
+
+    # If file exists save the journal
+    if [ -e $logfile ]; then
+        # Add a separator
+        echo "### LEAPP reboot ###" > $logfile
+
+        # write out the logfile
+        journalctl -a -m >> $logfile
+
+        # We need to run the actual saving of leapp-upgrade.log in a container and mount everything before, to be
+        # sure /var/log is mounted in case it is on a separate partition.
+        local store_cmd="mount -a"
+        local store_cmd="$store_cmd; cat /tmp-leapp-upgrade.log >> /var/log/leapp/leapp-upgrade.log"
+
+        /usr/bin/systemd-nspawn $NSPAWN_OPTS -D "$NEWROOT" /usr/bin/bash -c "$store_cmd"
+
+        rm -f $logfile
+    fi
+}
+
+
+############################### MAIN #########################################
+
+# workaround to replace the live root arg by the host's real root in
+# /proc/cmdline that is read by /usr/lib/kernel/50-dracut.install
+# during the kernel-core rpm postscript.
+# the result is ro-bind-mounted over /proc/cmdline inside the container.
+awk '{print $1}' /proc/cmdline \
+    | xargs -I@ echo @ $(cat "${NEWROOT}"/var/lib/leapp/.fakerootfs) \
+    > ${NEWROOT}/var/lib/leapp/.fakecmdline
+
+##### do the upgrade #######
+(
+    # check if leapp previously failed in the initramfs, if it did return to the emergency shell
+    [ -f "$NEWROOT$LEAPP_FAILED_FLAG_FILE" ] && {
+        echo >&2 "Found file $NEWROOT$LEAPP_FAILED_FLAG_FILE"
+        echo >&2 "Error: Leapp previously failed and cannot continue, returning back to emergency shell"
+        echo >&2 "Please file a support case with $NEWROOT/var/log/leapp/leapp-upgrade.log attached"
+        echo >&2 "To rerun the upgrade upon exiting the dracut shell remove the $NEWROOT$LEAPP_FAILED_FLAG_FILE file"
+        exit 1
+    }
```

Comment:
the shell script does not reflect some recent upstream changes requested by support. e.g. this should not block the upgrade (even when it will most likely lead to crash later). As this is experimental only, it's not a problem, but we should reflect the changes at least in a follow up PR